src/RoboSaverVirtualModule.sol

```diff
@@ -304,7 +336,13 @@ contract RoboSaverVirtualModule is
         } else if (_action == VirtualModule.PoolAction.DEPOSIT) {
             _poolDeposit(_amount);
         } else if (_action == VirtualModule.PoolAction.CLOSE) {
-            _poolClose(_amount);
+            if (BPT_STEUR_EURE.balanceOf(CARD) == 0) {
+                _unstakeAndClaim();
```

Comment:
overall everything looks good and great coverage on the tests, only im not sure why you decided here to do this on multi-steps?

rather than batching in one go `_unstakeAndClaim` & `_poolClose` in a multicall?. since the `bptBalance` has equality w/ whatever is on the gauge itself `gaugeBalance`, right?

for example `_unstakeAndClaim` could return the rewards pool balance from L454 and use it as a conditional argument for pool closure on the `poolClose` method in such a way these actions could be done atomically

wdyt?